Tools/Scripts/layout-test-failure-analysis

```diff
@@ -0,0 +1,291 @@
+# Returns true if it exists at the end of the attempt, false if not.
+def ensure_directory_exists(dir):
+    if not os.path.isdir(dir):
+        os.makedirs(dir, exist_ok=True)
+        if not os.path.isdir(dir):
+            return False
+    return True
+
+
+# Takes a string and exports to specified file path.
+# Prompts the user for a new file path if there's an IO error.
+def export_file(path, data, no_warn = False):
+    path
+    while True:
+        try:
+            path = os.path.abspath(os.path.expanduser(path))
+            dir = os.path.dirname(path)
+            if os.path.isfile(path) and not no_warn: # file exists
+                choice = input(f'A file already exists at the location {path}. Is it okay to overwrite it? (Y/[N]): ').lower()
+                if choice.lower() != 'y' and choice.lower() != 'yes':
+                    raise ValueError()
+            if not ensure_directory_exists(dir): # directory doesn't exist
+                warning(f'Directory does not exist at {dir} and an error occurred while attempting to create it. Check permissions/spelling and try again.')
+                raise IOError()
+
+            # write file
+            print(path)
+            f = open(path, "w")
+            f.write(data)
+            f.close()
+            return path
+        except IsADirectoryError:
+            warning(f'The specified path "{path}" is a directory. Try entering a new file path.')
+        except OSError:
+            warning(f'The specified location "{path}" is read-only. Try entering a new file path.')
+        except ValueError as e:
+            pass
+        except Exception as e:
+            warning(f'Unable to write file: {e.args[1]}. Try entering a new file path.')
+
+        # only executes if file not written
+        path = input(f'New file path (enter nothing to quit): ')
+        print(path)
+        if path == '':
+            error('User quit without writing file.')
+            graceful_exit(os.EX_DATAERR)
+
+
+# Takes a dict of workers' test lists and exports the requested one(s) to a specified file path.
+def export_worker(path, data, num=-1, no_warn = False):
+    path = os.path.abspath(os.path.expanduser(path))
+    dir, filename = os.path.split(path)
+    if num > -1: # one worker
+        info(f'Exporting test run log from worker {num}...')
+        if not os.path.isfile(path) and ensure_directory_exists(dir) and os.path.isdir(path):
+            path += f'/worker_{num}_tests.txt' # if not a file, directory exists, and path is a directory, add a standard file name
+        exp = data.get(f'{num}')
+        export_file(path, '\n'.join(exp), no_warn)
+        info(f'Finished exporting worker {num}\'s test run log to "{path}".')
+    else: # all workers
+        num_workers = len(data.keys())
+        if not no_warn:
+            choice = input(f'{num_workers} worker file(s) are about to be output. Is it okay to bypass individual file overwrite confirmation messages? (Y/[N]): ').lower()
```

Comment:
Since you need to import `webkitpy` anyways, might consider `webkitcorepy`'s `Terminal.choose` here.

Reply:
Nifty!